### PR TITLE
Allow region to be configured

### LINKED ISSUE #8: [Feature] Support region option when intializing Mailgun
Hey mate!

thanks for making this lib!

Mailgun currently supports setting up domains in specific regions, which means you have to set it up differently. For example the URL scheme for a EU domain is `https://api.eu.mailgun.net/v3/<DOMAIN>`. Currently the base domain is hardcoded, which means you can't configure it.

Would be great if one could give either a different base domain or add a region when initializing mailgun.

If you'd like I can try to implement it and make a pull request.

Regards

### PR DESCRIPTION
**Pull request by maexrakete**
Fixes #8

### CODE REVIEW

**Comment by maexrakete** [comment]
I think you need to re-run this travis build, obviously it couldn't fetch a library...

File: src/main/scala/org/matthicks/mailgun/Mailgun.scala [hunk 1]
```diff
@@ -15,12 +15,16 @@ import scala.concurrent.Future
 import scala.concurrent.ExecutionContext.Implicits.global
 import io.youi.net._
 
-class Mailgun(domain: String, apiKey: String) {
+class Mailgun(domain: String, apiKey: String, regionOpt: Option[String] = None) {
```
Comment (darkfrog26) [inline src/main/scala/org/matthicks/mailgun/Mailgun.scala]: Maybe better named just `region`

File: src/main/scala/org/matthicks/mailgun/Mailgun.scala [hunk 2]
```diff
@@ -15,12 +15,16 @@ import scala.concurrent.Future
...
   private implicit val customConfig: Configuration = Configuration.default.withSnakeCaseMemberNames.withDefaults
+  private val regionDomain = regionOpt match {
+    case Some(region) => s"https://api.$region.mailgun.net/v3/$domain/messages"
+    case None => s"https://api.mailgun.net/v3/$domain/messages"
+  }
 
   private lazy val encodedKey = new String(Base64.getEncoder.encode(s"api:$apiKey".getBytes(StandardCharsets.UTF_8)), "utf-8")
   private lazy val client = HttpClient
-    .url(URL(s"https://api.mailgun.net/v3/$domain/messages"))
+    .url(URL(regionDomain))
```
Comment (darkfrog26) [inline src/main/scala/org/matthicks/mailgun/Mailgun.scala]: Maybe change this to be something like:
```scala
private lazy val url: URL = URL(s"https://api.${region.map(r => s"$r.").getOrElse("")}mailgun.net/v3/$domain/messages")
private lazy val client: HttpClient = HttpClient.url(url)
```

**Comment by maexrakete** [comment]
There you go!

**Comment by darkfrog26** [comment]
Great work!

**Comment by darkfrog26** [comment]
Any rush on a new release for this?

**Comment by maexrakete** [comment]
Would be great, I'm depending on this a little bit

**Comment by darkfrog26** [comment]
Released 1.0.12

**Pull request merged**